iOS Client/Store/MBPodcastsStore.swift

```diff
@@ -133,18 +133,26 @@ class MBPodcastsStore: PodcastsRepository {
         return self.streams
     }
     
-    func getVisibleStreams() -> [PodcastStream] {
+    func getVisibleFilterOptions() -> [PodcastFilterOption] {
         guard let visibilitySettings = UserDefaults.standard.object(forKey: streamVisibilityDefaultsKey) as? [String: Bool] else {
             return []
         }
         
-        return visibilitySettings.keys.compactMap { PodcastStream(rawValue: $0)}.filter { visibilitySettings[$0.rawValue] ?? false}
+        return visibilitySettings.keys.compactMap {
+            if let stream = PodcastStream(rawValue: $0) {
+                return PodcastFilterOption.stream(stream)
+            } else if $0 == PodcastFilterOption.downloaded.key {
+                return PodcastFilterOption.downloaded
+            } else {
+                return nil
+            }
+        }.filter { visibilitySettings[$0.key] ?? false}
```

Comment:
it would make more sense to filter first I think

Reply:
Actually I don't think we really even need the filter at all here. If the item in the array makes it past the compactMap then of course it's going to be in the visibilitySettings